Extension/src/LanguageServer/extension.ts

```diff
@@ -1162,3 +1162,39 @@ export function UpdateInsidersAccess(): void {
         void vscode.commands.executeCommand("workbench.extensions.installExtension", "ms-vscode.cpptools", { installPreReleaseVersion: true }).then(undefined, logAndReturn.undefined);
     }
 }
+
+export async function PreReleaseCheck(): Promise<void> {
+    // First we need to make sure the user isn't already on a pre-release version.
+    if (!util.packageJson.__metadata?.preRelease === true) {
+        // Get the info on the latest version from the marketplace to check if there is a pre-release version available.
+        const response = await fetch('https://marketplace.visualstudio.com/_apis/public/gallery/extensionquery', {
+            method: 'POST',
+            headers: {
+                Accept : 'application/json; api-version=3.0-preview',
+                'Content-Type' : 'application/json',
+            }, 
+            body: '{"filters": [{"criteria": [{"filterType": 7, "value": "ms-vscode.cmake-tools"}]}], "flags": 529}'
+        });
+        
+        const data = await response.json();
+        const preReleaseAvailable = data.results[0].extensions[0].versions[0].properties.some((e: Object) => Object.values(e).includes("Microsoft.VisualStudio.Code.PreRelease"));
+
+        // If the user isn't on the pre-release version, but one is available, prompt them to install it.
+        if (preReleaseAvailable) {
+            const message: string = localize("prerelease.message", "There is a new pre-release version of the C/C++ extension, would you like to install it?");
+            const yes: string = localize("yes.button", "Yes");
+            const no: string = localize("no.button", "No");
+            vscode.window.showInformationMessage(message, yes, no).then((selection) => {
+                switch (selection) {
```

Comment:
I believe that since you're just comparing selection to yes, you can use an if statement. Is there a reason for the default case?